pkg/components/metadata.go

```diff
@@ -116,51 +34,12 @@ func LabelValueOneOf(labels map[string]string, componentTypes ...ComponentType)
 	return false, nil
 }
 
-// Equal checks if the other is a ComponentType or some type that can be parsed
-// and then checks for equality.
-func (ct ComponentType) Equal(other any) (bool, error) {
-	switch o := other.(type) {
-	case string:
-		parsed, err := ToComponentType(o)
-		if err != nil {
-			return false, err
-		}
-		return parsed == ct, nil
-	case ComponentType:
-		return ct == o, nil
-	case fmt.Stringer:
-		return ct.Equal(o.String())
-	default:
-		return false, errors.Errorf("unknown component type")
-	}
-}
-
-// MarshalJSON marshals a `ComponentType` into JSON bytes
-func (ct ComponentType) MarshalJSON() ([]byte, error) {
-	return []byte(`"` + ct.String() + `"`), nil
-}
-
-// MarshalText marshals the `ComponentType` into text bytes
-func (ct ComponentType) MarshalText() ([]byte, error) {
-	return []byte(`"` + ct.String() + `"`), nil
-}
-
-// UnmarshalJSON unmarshalls bytes into a `ComponentType`
-func (ct *ComponentType) UnmarshalJSON(b []byte) error {
-	b = bytes.Trim(bytes.Trim(b, `"`), ` `)
-	parsedComponent, err := ToComponentType(string(b))
-	if err == nil {
-		*ct = parsedComponent
-	}
-	return err
-}
-
-// UnmarshalText unmarshalls bytes into a `ComponentType`
-func (ct *ComponentType) UnmarshalText(text []byte) error {
-	text = bytes.Trim(bytes.Trim(text, `"`), ` `)
-	parsedComponent, err := ToComponentType(string(text))
-	if err == nil {
-		*ct = parsedComponent
+// MustGetComponentType converts a string into a ComponentType enum value or
+// panics
+func MustGetComponentType(cts string) ComponentType {
```

Comment:
I don't think we need this any more, the tool has produced a `MustParseComponentType` that does the same thing